spec/compiler/crystal/tools/playground_spec.cr

```diff
@@ -469,7 +469,7 @@ describe Playground::AgentInstrumentorTransformer do
     end
     ), <<-CR
     def foo(x)
-      yield _p.i(3) { x }
```

Comment:
From `$ crystal play -v` i see that the instrumented code for 

```crystal
def foo(x)
  yield x
end
foo(3) do |a|
  a
end
```

is

```crystal
def foo(x)
  yield _p.i(2) do
    x
  end
end
_p.i(4) do
  foo(3) do |a|
    _p.i(5) do
      a
    end
  end
end
```

which does not compile. The to_s of the code uses `begin; x; end` instead of `{ x }` for the block.

So there is actually a bug in the `Crystal::ToSVisitor` for printing yields with calls with block. begin/end should be avoided and `{ ... }` should be used to preserve semantics OR a `begin/end` should wrap the call:

```crystal
def foo(x)
  yield begin
    _p.i(3) do
      x
    end
  end
end
``` 

I would rather not remove the instrumentation for single argument yields. The best thing would be to fix the ToSVisitor. The second best is to change the agent_instrumentor_transformer.cr to explicitly wrap the `instrument(...)` with a `begin/end`.